### PR TITLE
[Keyboard] Add Keychron K2v2 RGB ISO

### PR DESCRIPTION
**Pull request by KeijoMika**
## Description

Add Keychron K2v2 RGB ISO keyboard to sn32_develop

Based on previous work of @dexter93 on the K2, @PythonDeployer and @ephb on the K4, and @tbowmo on the K8. Guidance also provided by @karliss.

I just transferred some settings to info.json and added VIA support. Layout file also provided for VIA.

## Supported
- [x] RGB
- [x] VIA
- [ ] Full original functionality (Bluetooth not working)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2023 KeijoMika (https://github.com/KeijoMika)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
```
Comment (dexter93) [inline keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h]:
```suggestion
```
default is 4. no need to declare again

File: keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+#define VIA_QMK_RGBLIGHT_ENABLE
```
Comment (dexter93) [inline keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h]:
```suggestion
```
this flag is obsolete. it's been removed.

**Review by dexter93 — CHANGES_REQUESTED** [review]

File: keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h (hunk 2 above)
Comment (dexter93) [reply keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h]: the entire file can be safely removed

File: keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h (hunk 1 above)
Comment (KeijoMika) [reply keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h]: Thank you. Removed keyboards/keychron/k2/rgb/v2/iso/keymaps/via/config.h

File: keyboards/keychron/k2/rgb/v2/iso/readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 1 way:
+
+* **Physical Bootloader Pins**: Turn off the keyboard, Using a pair of tweezers or a paperclip, short the 2 pins left from spacebar switch holes and hold the tweezers on the pins while turning the keyboard on, this should put it into the bootloader.
```
Comment (dexter93) [inline keyboards/keychron/k2/rgb/v2/iso/readme.md]:
```suggestion
Enter the bootloader in 2 ways:

* **Physical Bootloader Pins**: Turn off the keyboard, Using a pair of tweezers or a paperclip, short the 2 pins left from spacebar switch holes and hold the tweezers on the pins while turning the keyboard on, this should put it into the bootloader.
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available. In the default keymap Fn+Esc.
```

**Review by dexter93 — CHANGES_REQUESTED** [review]

**Pull request merged**